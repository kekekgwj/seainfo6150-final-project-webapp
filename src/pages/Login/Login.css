.login-box {
    border: 1px solid #d8d8d8;
    width: 340px;
    position: relative;
    margin-top: -30px;
    vertical-align: top;
}
.login-wrapper {
    width: 1000px;
    margin: 20px auto 0;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
}
.login-img {
    overflow: visible;
    margin-top: 50px;
}


.account-body {
    min-height: 300px;
    width: 340px;
    padding: 50px 30px 100px;
    overflow: hidden;
    position: relative;
    background: #fff;
    box-sizing: border-box
}

.account-title {
    margin-bottom: 10px;
    color: #333;
    border-bottom: 1px solid #ececec;
    font-size: 18px;
    line-height: 2;
    text-align: center;
}

.account-form-error {
    min-height: 20px;
    margin-bottom: 5px;
    background: none;
    font-size: 13px;
    color: red;
}

.account-form-raw {
    margin-bottom: 10px;
}

.account-form-input {
    width: 280px;
    font-size: 13px;
    padding: 10px;
    border-radius: 3px;
    box-sizing: border-box;
    border: 1px solid #e4e6e5;
}
.account-form-field-submit {
    width: 280px;
}
.btn-account {
    line-height: 36px;
    text-align: center;
    outline: 0;
    cursor: default;
    background-color: #b92b27;
    border-radius: 3px;
    color: #fff;
    height: 36px;
}

.loading{
    width: 150px;
    height: 15px;
    margin: 0 auto;
    margin-top:100px;
}
.loading span{
    display: inline-block;
    width: 15px;
    height: 100%;
    margin-right: 5px;
    border-radius: 50%;
    background: #ddd;
    animation: load 1.04s ease infinite;
}
.loading span:last-child{
    margin-right: 0px; 
}
@keyframes load{
    0%{
        opacity: 1;
    }
    100%{
        opacity: 0;
    }
}
.loading span:nth-child(1){
    animation-delay:0.13s;
}
.loading span:nth-child(2){
    animation-delay:0.26s;
}
.loading span:nth-child(3){
    animation-delay:0.39s;
}
.loading span:nth-child(4){
    animation-delay:0.52s;
}
.loading span:nth-child(5){
    animation-delay:0.65s;
}