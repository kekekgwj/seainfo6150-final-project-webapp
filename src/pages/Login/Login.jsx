import React, { useEffect } from "react";
import "./Login.css"
import { useState } from "react";
import { performLogin } from "../../services"
import { successMsgs, errMsgs} from "../../base";
import { useHistory, useLocation } from 'react-router-dom';
import Movie200 from '../../assets/movie-200.jpg';
import Movie400 from '../../assets/movie-400.jpg';
import Movie500 from '../..//assets/movie-500.jpg';

const Login = (props) => {
    const {
        onLogin,
    } = props;
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [isLoading, setIsLoading] = useState(false);
    const [inputValid, setInputValid] = useState(false);
    const [status, setStatus] = useState('');
    const history = useHistory();
    const location = useLocation();

     useEffect(() => {
        if (!username || !password) {
            setInputValid(false);
        }
        else {
            setInputValid(true);
        };
    }, [username, password])
    const getLogin = () => {
        if (isLoading) {
            return (
            <div className="loading">
                    <span></span>
                    <span></span>
                    <span></span>
                    <span></span>
                    <span></span>
            </div>)
        }
        else {
            return (<span>Login</span>)
        }
    }
    const login = () => {
        if (!inputValid) {
            setStatus('username / password can not be empty');
            return;
        }
        setStatus('');
        setIsLoading(true);
        performLogin({username, password})
        .then( () => {
            setStatus( successMsgs['login-success'] );
            setIsLoading(false);
            onLogin(username);
            setTimeout(() => {afterLoginJump()}, 2000);
        })
        .catch( err => {
            setStatus(errMsgs[err.error] || err.error);
            setIsLoading(false);
        })
    }
    const afterLoginJump = () => {
        if (location.state && location.state.nextPathname && location.state.pageInfo ) {
            history.push({pathname:location.state.nextPathname, state:{id: location.state.pageInfo.id, name: location.state.pageInfo.name}})
        }
        else {
            history.push('/movie_list');
        }
       
    }
    return (
        <div className="login-wrapper">
            <div className="login-img">   
                <picture>
                    <source 
                        media="(max-width:800px)"
                        srcset={Movie200}>
                    </source>
                    <source 
                        media="(max-width:1100px)"
                        srcset={Movie400}>
                    </source>
                    <source 
                        media="(max-width:1500px)"
                        srcset={Movie500}>
                    </source>
                    <img src={Movie500} alt="This login movie img"></img>
                </picture>
            </div>
            <div className="login-box"> 
                <div className="account-body">
                    <div className="account-title">
                        Login
                    </div>
                    <div className="account-form-wrapper">
                        <div className="account-form">
                            <div className="account-form-raw">
                                <input 
                                id="username" name="username" 
                                type="text" className="account-form-input" 
                                placeholder="username" 
                                value={username} onChange={e => setUsername(e.target.value)}></input>
                            </div>
                            <div className="account-form-raw">
                                <input id="password" name="password" type="text" className="account-form-input" placeholder="password" value={password} onChange={e => setPassword(e.target.value)}></input>
                            </div>
                            <div className="account-form-error">
                                <span className="error-message">{status}</span>
                            </div>
                            <div className="account-form-field-submit" onClick={login}>
                                <div className={`btn-account ${isLoading ? "btn-loading" : null}`}>{getLogin()}</div>
      
                            </div>
                        </div>
                    </div>
                </div>
            </div>

        </div>)
} 

export default Login;