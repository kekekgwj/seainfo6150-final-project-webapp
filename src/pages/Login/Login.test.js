import React from "react";
import Login from "./Login";

describe("Login page tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Login onLogin={() => {console.log('test onLogin func')}}/>);
    expect(container).toMatchSnapshot();
  });
});