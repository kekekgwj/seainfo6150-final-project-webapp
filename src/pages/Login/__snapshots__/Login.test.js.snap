// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page tests renders correctly 1`] = `
<div>
  <div
    class="login-wrapper"
  >
    <div
      class="login-img"
    >
      <picture>
        <source
          media="(max-width:800px)"
          srcset="movie-200.jpg"
        />
        <source
          media="(max-width:1100px)"
          srcset="movie-400.jpg"
        />
        <source
          media="(max-width:1500px)"
          srcset="movie-500.jpg"
        />
        <img
          alt="This login movie img"
          src="movie-500.jpg"
        />
      </picture>
    </div>
    <div
      class="login-box"
    >
      <div
        class="account-body"
      >
        <div
          class="account-title"
        >
          Login
        </div>
        <div
          class="account-form-wrapper"
        >
          <div
            class="account-form"
          >
            <div
              class="account-form-raw"
            >
              <input
                class="account-form-input"
                id="username"
                name="username"
                placeholder="username"
                type="text"
                value=""
              />
            </div>
            <div
              class="account-form-raw"
            >
              <input
                class="account-form-input"
                id="password"
                name="password"
                placeholder="password"
                type="text"
                value=""
              />
            </div>
            <div
              class="account-form-error"
            >
              <span
                class="error-message"
              />
            </div>
            <div
              class="account-form-field-submit"
            >
              <div
                class="btn-account null"
              >
                <span>
                  Login
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
